Add error_message for deduplicated parts in system.part_log

AVMusorin opened the pull request:
<!---
A technical comment, you are free to remove or leave it as it is when PR is created
The following categories are used in the next scripts, update them accordingly
utils/changelog/changelog.py
tests/ci/cancel_and_rerun_workflow_lambda/app.py
-->
### Changelog category (leave one):
- Improvement


### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add error message that the part was deduplicated

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

<!---
Directly edit documentation source files in the "docs" folder with the same pull-request as code changes

or

Add a user-readable short description of the changes that should be added to docs.clickhouse.com below.

At a minimum, the following information should be added (but add more as needed).
- Motivation: Why is this function, table engine, etc. useful to ClickHouse users?

- Parameters: If the feature being added takes arguments, options or is influenced by settings, please list them below with a brief explanation.

- Example use: A query or command.
-->

yariks5s commented:
I propose to check this new logic with an integration test because I have concerns about this check, and I'm sure that this check you've added will eventually become broken/flaky in case we have an extensive testing process, and current test will be at the end of the testing queue, it already happens sometimes: 
```Code: 158. DB::Exception: Limit for rows (controlled by 'max_rows_to_read' setting) exceeded, max rows: 20.00 million, current rows: 134.60 million. (TOO_MANY_ROWS) (version 25.4.1.36938 (official build)) (from [::1]:60320) (comment: 02124_insert_deduplication_token_replica.sql)```

This is because we may have allocated a lot of logging entries in the table, and then we're unable to execute the query due to the CI restrictions. Checking this simple logic with an integration test can solve this problem.

azat commented:
>I propose to check this new logic with an integration test because I have concerns about this check, and I'm sure that this check you've added will eventually become broken/flaky in case we have an extensive testing process, and current test will be at the end of the testing queue, it already happens sometimes:

It should be OK I think, the mentioned failure was caused by a different issue (besides we already have similar tests)
But we need to add filter by `database` here.

Also we need to add some changes for SMT likely

azat commented on tests/queries/0_stateless/02124_insert_deduplication_token_replica.sql:
```diff
@@ -13,6 +13,12 @@ INSERT INTO insert_dedup_token1 VALUES(1, 1001);
+SYSTEM FLUSH LOGS system.part_log;
+SELECT DISTINCT exception FROM system.part_log
+WHERE table = 'insert_dedup_token1'
+  AND event_type = 'NewPart'
```
```suggestion
  AND event_type = 'NewPart'
  AND database = currentDatabase()
```

AVMusorin replied:
Added database and table filters.

AVMusorin commented:
@yariks5s could we please continue this PR?

azat commented:
Looks good, but the private should be adjusted as well

AVMusorin commented:
@yariks5s don't you mind to continue?

AVMusorin committed:
Add error_message for deduplicated parts in system.part_log

AVMusorin committed:
tests: add table and database for 02124_insert_deduplication_token_replica

Fixes:

```
Code: 158. DB::Exception: Limit for rows (controlled by 'max_rows_to_read' setting) exceeded, max rows: 20.00 million, current rows: 134.60 million. (TOO_MANY_ROWS) (version 25.4.1.36938 (official build)) (from [::1]:60320) (comment: 02124_insert_deduplication_token_replica.sql)
```

Co-authored-by: Azat Khuzhin <a3at.mail@gmail.com>

AVMusorin commented:
@yariks5s Hey! just a friendly reminder about this PR
